tutorials/securing-gcs-static-website/index.md

```diff
@@ -0,0 +1,455 @@
+---
+title: Securing a static website hosted on Google Cloud Storage 
+description: This tutorial describes a solution to protect static assets on Cloud Storage based on the GCLB Serverless Network Endpoint Groups (NEGs).
+author: shenxiang
+tags: Serverless, Cloud Storage, GCLB, Network Endpoint Groups
+date_published: 2020-08-26
+---
+
+It's easy to create and maintain an[ HTTPS-based static website on Google Cloud Storage](https://cloud.google.com/storage/docs/hosting-static-website)(GCS) with Cloud CDN, Google Cloud Load Balancer(GCLB), managed SSL certificates, and custom domains. This serverless approach becomes popular due to the flexibility, scalability, and low cost.   
+However, it's still challenging to provide authentication and authorization for a static website on GCP **in a serverless fashion**.  This tutorial describes a solution to protect static assets on GCS based on the GCLB [Serverless Network Endpoint Groups](https://cloud.google.com/load-balancing/docs/negs/setting-up-serverless-negs) (NEGs).  
+Below is the high-level architecture of this solution:
+
+![gcs-static-arch](static-website/src/assets/arch-gcs-site.png)
+
+The architecture above incorporates the following key features:
+
+1. Create a load balancer to use a custom domain with a managed SSL certificate. When a request is received, we use [routing rules](https://cloud.google.com/load-balancing/docs/https/setting-up-query-and-header-routing#http-header-based-routing) to check whether [a signed cookie for Cloud CDN](https://cloud.google.com/cdn/docs/private-content#signed_cookies) is in the request header. If the cookie doesn't exist, redirect the request to a login app hosted on Cloud Run. If there is a signed cookie, send the request to the Cloud CDN of the backend bucket.
+
+1. The login app performs authentication. If the authentication is successful, the login app [generates a signed cookie ](https://cloud.google.com/cdn/docs/using-signed-cookies)and sends it back to the client with an HTTP redirect. The cookie works because the login app and the bucket backends are behind the same load balancer. Therefore, they are considered as **the [same origin](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) (Important!)**. Even if a user can access the default Cloud Run endpoint directly and log in from there, she still doesn't have access to the CDN since the cookie is not from the same origin.
+
+1. The CDN [signed cookie is verified by Cloud CDN](https://cloud.google.com/cdn/docs/using-signed-cookies), providing access to the static assets. We can specify an expiration time for the cookie. Once it's expired, modern browsers either delete the cookie or stop sending it. Cloud CDN will also reject expired cookies.
+
+1. Permit Cloud CDN to read the objects in our private bucket by adding the Cloud CDN service account to Cloud Storage's ACLs.
+
+## Objectives 
+
+-  Learn to build and deploy a static website to a Cloud Storage bucket.
```

Comment:
IMO remove the repetitive "Learn to" from these, as at the end of the tutorial, you've actually done them, not just learned